### DISCUSSION TITLE
Birt "page control"

### DISCUSSION THREAD

**Discussion by Jonasbr93** [eclipse-birt/birt#831] [Q&A]
Hello once again, im making a report and im trying to make a report thats has basicaly 2 pages. And the problem starts here. 

The second page the "layout" of the page is always the same but for the First one it will change like eveything (on the layout) - also im doing a page break at the moment inside of the grid (so i only have one master page) -  i know that its possible to hide a field in the propertys and make a if statment  that will show or not that field. But is it possible aswell that for example to make that field move like to the center of the page (example) 

example : i have a grid with 2 colums - colum1 - field_a   colum2-fiueld _b    

if (row["fielda"] != null ){
false
}else {
true
}
--> so if field_a is filled with somenthing diferente than null it wil be show else if its equal null it will "disapear" and if it disapears will it be possible to move the field_b to the center of the layout?

also i dont think even if that was possible  to be done it would be enough. to solve my problems, because i will have many diferent fields in the other layouts...

If you have any suggestions or example to help me sort this out it would be amazing.

thanks in advance

**Comment by Jonasbr93** [comment]
actually i think ive found a solution : inside the same table  i will make differets grid for each "layout" and than i will control the grids visibility. Now i need to just figure out how its is possible to make a if statment checking if a field is null. For some reason its not working 

if (row._outer["DOMAIN"] == 'X' && row._outer["INSTRUCTOR_NAME"] == null){
false	
}else {
true
}

Any help?

**Reply by colinsutton** [reply]
Your overall layout is not clear to me but maybe you want to set the visibility rule on the first column. Should the 'X' be "X"?

**Comment by hvbtup** [comment]
You can also simplify your expressions. Note that "!" means "not". So your expression is equivalent to

    ! (row._outer["DOMAIN"] == 'X' && row._outer["INSTRUCTOR_NAME"] == null)

And the right half of the && can be simplified to 

    ! row._outer["INSTRUCTOR_NAME"]

This is not equivalent, but probably is what you mean. For a string value `s`, `!s` is true if `s` is null or `s` is an empty string.

As a second note, you can hide not only rows, but also columns of a grid/table by using an visibility expression.

However, I don't really understand the problem you want to solve.

**Comment by TheLaserNinja** [comment]
Sounds like a Visibility expression. You may also need to add a script in one of the elements events to do the move. Without seeing the layout of what you are trying to do it's hard to say.
